### DISCUSSION TITLE
Saleor Dashboard deployment to Vercel / Netlify

### DISCUSSION THREAD

**Discussion by Horib** [mirumee/saleor-dashboard#2446] [Q&A] [answered]
Hey guys and gals, 

Trying to deploy Saloer Dashboard to production, with both Vercel and Netlify, It works ok, I can login, add, edit etc. in the dashboard... but. If an Action requires I open a link e.g. vercel.example.com/configuration, OR i refresh i get a 404 Page not Found.

Saleor Dashboard works when i run it locally, that is on localhost:9000, I have no problems refreshing or opening URLs further into the dashboard.

My conclusion right now is, that it has something to do with how both Vercel and Netlify build the project.
I have tried googling, changing the build commands etc., but I can't figure out how to fix the problem.

I should mention, i have added the environment variable **API_URI** which links to my Saleor Core Server

**Deployment on Vercel**
![msedge_YlE6V3efMa](https://user-images.githubusercontent.com/25446543/197407298-0842862a-d174-4a60-b12d-3e90a781bc3f.png)

Got to the link or refresh and i get this error
![msedge_oTBUQmJ2nR](https://user-images.githubusercontent.com/25446543/197407231-6926cbad-fc47-4351-aa74-5ce604525415.png)

**Vercel build configuration**
![msedge_VldDsB6Opd](https://user-images.githubusercontent.com/25446543/197407468-8d73f46d-79df-475d-9a64-28ab10168e1d.png)

**Deployment on Netlify**
![msedge_AMNbgMT2ek](https://user-images.githubusercontent.com/25446543/197407891-b8e10bc1-ba7f-49bf-8aeb-be6054409f45.png)

Got to the link or refresh and i get this error
![msedge_vemMHiKjc4](https://user-images.githubusercontent.com/25446543/197407253-fce39911-e0ff-4d2a-bf6b-4e392c26534d.png)

**Netlify build configuration**
![msedge_uEsZkz3pqU](https://user-images.githubusercontent.com/25446543/197407460-0a47fc81-5553-458d-8212-e75438038454.png)

**Comment by magul** [comment]
Hi @Horib , have you set 404 custom page?

Dashboard uses React Router under the hood to render different pages under different paths, but that's done exclusively in browser (as this path doesn't exist as something routable on a server). In that situation you have to redirect all 404s your server is getting to `index.html`, that way it's you browser will bootstrap Dashboard (as React app) and will handle that route on client side.

**Comment by andrzejewsky** [comment, marked as answer]
I think it's because vercel doesn't know that you have an SPA app. In the SPA we need to tell the server that all traffic must go to the index of the app. It's the same for any reverse-proxy / web-servers, otherwise, a web-server will try to match based on the filesystem.

Here is more info https://vercel.com/docs/project-configuration#project/rewrites I think for you will work (vercel.json):

```json
{
  "rewrites":  [
    {"source": "/(.*)", "destination": "/"}
  ]
}
```
So you basically rewrite everything in the URL always to `/` which is understandable by the server (index), where also SPA lives.

**Reply by Horib** [reply]
Fantastic it works :) thank you very much
